pinot-broker/src/main/java/org/apache/pinot/broker/broker/AccessControlFactory.java

```diff
@@ -28,11 +30,24 @@ public abstract class AccessControlFactory {
   public static final Logger LOGGER = LoggerFactory.getLogger(AccessControlFactory.class);
   public static final String ACCESS_CONTROL_CLASS_CONFIG = "class";
 
-  public abstract void init(PinotConfiguration confguration);
+  public void init(PinotConfiguration configuration) {
+  };
+
+  /**
+   * Extend original init method inorder to support Zookeeper BasicAuthAccessControlFactory
```

Comment:
One issue I see with this approach is that we will end up adding flavors of `init()` for each of the auth factories depending on what they needed. One alternative might be to have something like `PinotHelixResourceManager` that can give access to cluster level properties, to make it more shareable across other future factory inits.

Reply:
@mayankshriv Sorry, I don't get your point. In my design, the reason why I pass ZkHelixProperStore or something familiar parameters is just used to init class AccessControlUserCache to keep track of user config other than get some cluster properties.

Reply:
I undertand. I am saying if we use `PinotHelixResourceManager` then you can still get it, and in future if there are other implementations, chances are they will also get the info they need from this class, as opposed to `ZKHelixPropertyStore`.

Reply:
+1 either use PinotHelixResourceManager or ZkHelixPropertyStore since you used PinotHelixResourceManager in the init method of `ZkBasicAuthAccessControlFactory`